### PR TITLE
Initial (minimal) tweaks after dask key type changes

### PR DESCRIPTION
**Pull request by crusaderky**
- Part of #8130
- Fix mypy errors after https://github.com/dask/dask/pull/10485
- Import `validate_key` from dask/dask

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -65,3 +65,6 @@ repos:
           - pyarrow
           - git+https://github.com/dask/dask
           - git+https://github.com/dask/zict
+
+# Increase this value to clear the cache on GitHub actions
+# if nothing else in this file has changed: 0
```
Comment (phofl) [inline .pre-commit-config.yaml:70]: Which value? That comment is a bit confusing

File: distributed/shuffle/_shuffle.py [hunk 2]
```diff
@@ -207,19 +207,19 @@ def _cull(self, parts_out: Iterable[int]) -> P2PShuffleLayer:
...
         """Simple utility to convert keys to partition indices."""
         parts = set()
         for key in keys:
             if isinstance(key, tuple) and len(key) == 2:
-                _name, _part = key
-                if _name != self.name:
-                    continue
-                parts.add(_part)
+                name, part = key
+                if part == self.name:
```
Comment (phofl) [inline distributed/shuffle/_shuffle.py:216]: Is this change intentional?

**Review by phofl — COMMENTED** [review]
couple of small comments

File: distributed/shuffle/_shuffle.py (hunk 2 above)
Comment (crusaderky) [reply distributed/shuffle/_shuffle.py:216]: yes (mostly cosmetic). The necessary bit is the assertion

File: .pre-commit-config.yaml (hunk 1 above)
Comment (crusaderky) [reply .pre-commit-config.yaml]: improved clarity

File: distributed/shuffle/_shuffle.py (hunk 2 above)
Comment (phofl) [reply distributed/shuffle/_shuffle.py:216]: Thx!

Comment (crusaderky) [reply distributed/shuffle/_shuffle.py:216]: whoops. Saw it now. Thanks for spotting!

**Review by phofl — APPROVED** [review]
